ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto

```diff
@@ -0,0 +1,165 @@
+//
+// Protocol buffer messages used by the participant state key-value utilities
+// for implementing a DAML ledger backed by a key-value store.
+//
+// These messages must only be produced and consumed by the methods in
+// `KeyValueCommitting`, `KeyValueConsumption` and `KeyValueSubmission` classes.
+// You must not embed these messages in other protocol buffer messages.
+//
+// TODO(JM): We're consistently using the 'Daml' prefix to allow unqualified import
+// of the generated classes and to distinguish these from the scala types in participant-state.
+// Might want to use a more telling prefix.
+
+syntax = "proto3";
+package com.daml.ledger.participant.state.kvutils;
+option java_package = "com.daml.ledger.participant.state.kvutils";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "da/daml_lf.proto";
+import "daml-lf/transaction/src/main/protobuf/transaction.proto";
+
+// The ledger implementation stores the `DamlLogEntry`s in an
```

Comment:
Hard to understand this comment without context. Please point to what context needs to be read before these comments start to make sense.